From patchwork Thu Dec 27 14:59:13 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [8/8] c: Don't error for const or restrict as asm-qualifier
X-Patchwork-Submitter: Segher Boessenkool <segher@kernel.crashing.org>
X-Patchwork-Id: 13825
Message-Id: <f0e4c2f3f89d0587a2ae547432d9a94bca9d06ff.1545922222.git.segher@kernel.crashing.org>
To: gcc-patches@gcc.gnu.org
Cc: Segher Boessenkool <segher@kernel.crashing.org>
Date: Thu, 27 Dec 2018 14:59:13 +0000
From: Segher Boessenkool <segher@kernel.crashing.org>
List-Id: <gcc-patches.gcc.gnu.org>

2018-12-27  Segher Boessenkool  <segher@kernel.crashing.org>

c/
	* c-parser.c (c_parser_asm_statement): Output a warning instead of an
	error for const and restrict.

testsuite/
	* gcc.dg/asm-qual-1.c: Adjust.

---
 gcc/c/c-parser.c                  | 2 +-
 gcc/testsuite/gcc.dg/asm-qual-1.c | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

-- 
1.8.3.1

diff --git a/gcc/c/c-parser.c b/gcc/c/c-parser.c
index 4baad62..a960169 100644
--- a/gcc/c/c-parser.c
+++ b/gcc/c/c-parser.c
@@ -6345,7 +6345,7 @@ c_parser_asm_statement (c_parser *parser)
 
 	case RID_CONST:
 	case RID_RESTRICT:
-	  error_at (loc, "%qE is not an asm qualifier", token->value);
+	  warning_at (loc, 0, "%qE is not an asm qualifier", token->value);
 	  c_parser_consume_token (parser);
 	  continue;
 
diff --git a/gcc/testsuite/gcc.dg/asm-qual-1.c b/gcc/testsuite/gcc.dg/asm-qual-1.c
index eff6b45..4982a6b 100644
--- a/gcc/testsuite/gcc.dg/asm-qual-1.c
+++ b/gcc/testsuite/gcc.dg/asm-qual-1.c
@@ -8,7 +8,7 @@ f (void)
 {
   asm volatile ("");
 
-  asm const (""); /* { dg-error {'const' is not an asm qualifier} } */
+  asm const (""); /* { dg-warning {'const' is not an asm qualifier} } */
 
-  asm restrict (""); /* { dg-error {'restrict' is not an asm qualifier} } */
+  asm restrict (""); /* { dg-warning {'restrict' is not an asm qualifier} } */
 }
